test/Benchmarks/Metrics/MetricsViewBenchmarks.cs

```diff
@@ -0,0 +1,104 @@
+
+
+|                    Method | ViewConfig |     Mean |    Error |   StdDev | Gen 0 | Gen 1 | Gen 2 | Allocated |
+|-------------------------- |----------- |---------:|---------:|---------:|------:|------:|------:|----------:|
+| CounterWith3LabelsHotPath |          1 | 514.3 ns | 10.32 ns | 17.24 ns |     - |     - |     - |         - |
+| CounterWith3LabelsHotPath |          2 | 549.1 ns | 10.98 ns | 23.17 ns |     - |     - |     - |         - |
+| CounterWith3LabelsHotPath |          3 | 540.1 ns | 10.74 ns | 13.58 ns |     - |     - |     - |         - |
+*/
+
+namespace Benchmarks.Metrics
+{
+    [MemoryDiagnoser]
+    public class MetricsViewBenchmarks
+    {
+        private static readonly ThreadLocal<Random> ThreadLocalRandom = new ThreadLocal<Random>(() => new Random());
+        private Counter<long> counter;
+        private MeterProvider provider;
+        private Meter meter;
+        private string[] dimensionValues = new string[] { "DimVal1", "DimVal2", "DimVal3", "DimVal4", "DimVal5", "DimVal6", "DimVal7", "DimVal8", "DimVal9", "DimVal10" };
+
+        [Params(1, 2, 3)]
+        public int ViewConfig { get; set; }
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            /* ViewConfig 1 = no views registered at all.
+             * ViewConfig 2 = view registered, but does not select the instrument
+             * ViewConfig 3 = view registed, selects the instrument with same tags as provided.
+             */
+
+            this.meter = new Meter("TestMeter");
+            this.counter = this.meter.CreateCounter<long>("counter");
+
+            if (this.ViewConfig == 1)
+            {
+                this.provider = Sdk.CreateMeterProviderBuilder()
+                    .AddSource(this.meter.Name)
+                    .Build();
+            }
+            else if (this.ViewConfig == 2)
+            {
+                this.provider = Sdk.CreateMeterProviderBuilder()
+                    .AddSource(this.meter.Name)
+                    .AddView(this.counter.Name + "notmatch", new MetricStreamConfiguration() { TagKeys = new string[] { "DimName1", "DimName2", "DimName3" } })
+                    .Build();
+            }
+            else if (this.ViewConfig == 3)
+            {
+                this.provider = Sdk.CreateMeterProviderBuilder()
+                    .AddSource(this.meter.Name)
+                    .AddView(this.counter.Name, new MetricStreamConfiguration() { TagKeys = new string[] { "DimName1", "DimName2", "DimName3" } })
+                    .Build();
+            }
+        }
+
+        [GlobalCleanup]
+        public void Cleanup()
+        {
+            this.meter?.Dispose();
+            this.provider?.Dispose();
+        }
+
+        [Benchmark]
+        public void CounterWith3LabelsHotPath()
+        {
+            var random = ThreadLocalRandom.Value;
+            var tag1 = new KeyValuePair<string, object>("DimName1", this.dimensionValues[random.Next(0, 10)]);
```

Comment:
nit: consider something like the following to improve readability:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/a4b25f117d599a4ba6ea0ac49f9423353730da55/test/OpenTelemetry.Tests.Stress.Metrics/Program.cs#L54